### PR TITLE
Redux FAD-4332

### PR DESCRIPTION
**Pull request by jasonrhodes**

### CODE REVIEW

File: src/pages/spf/components/SPFNode.js [hunk 1]
```diff
@@ -1,9 +1,22 @@
...
-function SPFNode({ domain = null, record, displayType = null, value = null, root = false, children = null }) {
+function SPFNode({
+  tree,
+  treeId,
+  domain = null,
+  record,
+  displayType = null,
+  value = null,
+  root = false,
+  children = null,
```
Comment (jonambas) [inline src/pages/spf/components/SPFNode.js]: I suggest we use `propTypes` and `defaultProps` for this kind of stuff. Its much easier to figure out whats going on looking at a component for the first time (for me at least).

File: src/reducers/spfInspect.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
+/* eslint-disable complexity */
+import moment from 'moment';
+import _ from 'lodash';
+
+const defaultState = {
+  results: { errors: [], warnings: [], spf_tree: { root: {}} },
+  loading: false,
+  error: null
```
Comment (jonambas) [inline src/reducers/spfInspect.js]: To decide which button to show ('Expand All' and 'Collapse All') - do we need one more 'expanded' prop here?

Comment (jonambas) [inline src/reducers/spfInspect.js]: actually nevermind, we can check root.expanded

**Comment by jasonrhodes** [comment]
We can't really choose a button because you can have mixed state, I thought it might be best to show both all the time.

File: src/pages/spf/components/SPFNode.js [hunk 3]
```diff
@@ -1,58 +1,78 @@
...
+  collapse
+}) {
   const label = domain ? domain : `${displayType}:${value}`;
+  const hasChildren = children && children.length > 0;
+
+  const onClick = () => {
+    if (!hasChildren) {
+      return null; // Prevent onClick on nodes w/ no children
+    }
+    return expanded ? collapse(treeId) : expand(treeId);
```
Comment (jonambas) [inline src/pages/spf/components/SPFNode.js]: Hey @jasonrhodes if you're wondering why I made this change - cause some nodes were returning empty children

screen shot 2017-02-02 at 10 18 08 am

File: src/pages/spf/components/SPFNode.js (hunk 1 above)
Comment (jasonrhodes) [inline src/pages/spf/components/SPFNode.js]: Yeah I might make this a class component again and connect it to redux ... 🤔

**Pull request merged**